az-cvm-vtpm/src/vtpm/mod.rs

```diff
@@ -165,7 +185,38 @@ pub fn get_quote(data: &[u8]) -> Result<Quote, QuoteError> {
     let signature = rsa_sig.signature().to_vec();
     let message = attest.marshall()?;
 
-    Ok(Quote { signature, message })
+    // Drop the context because access to the tpm device is again needed in the `read_all_pcrs` function.
+    drop(context);
+    let pcrs = read_all_pcrs()?;
+
+    Ok(Quote {
+        signature,
+        message,
+        pcrs,
+    })
+}
+
+/// Extract the 256 bank of PCRs from the TPM.
+fn read_all_pcrs() -> Result<Vec<Vec<u8>>, QuoteError> {
+    let mut pcrs: Vec<Vec<u8>> = Vec::new();
+
+    let conf: TctiNameConf = TctiNameConf::Device(DeviceConfig::default());
+    let mut context = Context::new(conf)?;
+    let hash_algo = HashingAlgorithm::Sha256;
+
+    // Reading one PCR slot at a time instead of feeding all the slots to the
+    // PcrSelectionListBuilder because `pcr_read` can only return limited results.
```

Comment:
there is `abstraction::pcr::read_all()` which provides a convenience fn for this